### ISSUE TITLE
Fullscreen applications don't see the correct output modes

### ISSUE THREAD

**Issue by pac85** [swaywm/sway#7942]
- **Sway Version:**
  - 1.8.1

- **Debug Log:**
  - https://gist.github.com/pac85/1835fa51b8f189fa0e4448e2bff32176

- **Configuration File:**
 default with the addition of the following lines
```
output HDMI-A-1 pos 0 0
output DP-1 pos 1920 0
output HDMI-A-2 pos 4480 0
```

- **Description:**

Fullscreen applications running on the `DP-1` display (which is 2560x1440@165hz) display see modes from the other displays (1920x1080@60hz). As a result some games glitch out, they appear in a smaller rectangle with black borders around them and the mouse mapping is wrong.
Only happens with XWayland applications.

![20240128161718_1](https://github.com/swaywm/sway/assets/8513550/2e5b9b34-6db5-4546-993f-29f00f5fcda7)

![s2](https://github.com/swaywm/sway/assets/8513550/038e6e6d-2478-401b-952f-e0c24f2c0e2e)

**Comment by zakk4223** [comment]
There's no X11 display flagged as primary, so some games do weird stuff like grab modes from the 0,0 screen.

Use xrandr to set a primary, xrandr --output DP-1 --primary

Like 80% of 'Xwayland is acting weird' bug reports are due to this. Has been for years.

Back when I used sway I had an IPC script that set it on every workspace change. You can't just set it once because if your monitors DPMS sleeps it may forget the setting. I also think if sway is doing lazy XWayland loading sometimes the process times out/goes away and when it respawns it will not have the primary set anymore.

To be fair I don't think there's a single wayland compositor that gets this correct.

**Comment by pac85** [comment]
Hey thanks that does the trick, not sure whether this issue should be closed though since I feel like this should be automatic somehow.

**Comment by zakk4223** [comment]
Yeah, it probably should be automatic/somehow configurable but at this point it's been an issue for so long that wine may get native wayland support soon and 90% of the apps that this impacts will get 'fixed'.

It looks like the next release of KDE/plasma has a way to designate 'display priority' and sets primary on the correct XWayland screen. I also wrote a hyprland plugin to fix it there.

**Comment by tank142** [comment]
"xrandr --output DP-1 --primary" does not solve the problem.
swaymsg -t get_outputs shows my monitors in reverse order:
```
Output HDMI-A-1 'Samsung Electric Company S24C36x H4PW601734'
  Current mode: 1920x1080 @ 74.973 Hz
Output DP-2 'Philips Consumer Electronics Company PHL 242E1GZ UK02052018154'
  Current mode: 1920x1080 @ 165.003 Hz
Output DP-1 'Philips Consumer Electronics Company 27M1N3500LS UK02241008808' (focused)
  Current mode: 2560x1440 @ 165.000 Hz
```
![2024-11-13-13-02-21](https://github.com/user-attachments/assets/2eea36e4-63ee-410d-a7e3-4cd006bbfdc5)
The only thing that helped me was turning off the monitors and turning them on again.
```
output {
	DP-2 disable
	HDMI-A-1 disable
}
exec_always swaymsg output DP-2 enable && swaymsg output HDMI-A-1 enable
```
```
Output DP-1 'Philips Consumer Electronics Company 27M1N3500LS UK02241008808' (focused)
  Current mode: 2560x1440 @ 165.000 Hz
Output DP-2 'Philips Consumer Electronics Company PHL 242E1GZ UK02052018154'
  Current mode: 1920x1080 @ 165.003 Hz
Output HDMI-A-1 'Samsung Electric Company S24C36x H4PW601734'
  Current mode: 1920x1080 @ 74.973 Hz
```
